test/jsonconfig-app.js

```diff
@@ -0,0 +1,31 @@
+import {app} from './support/app-config/app-config.js';
+import File from '../lib/file.js';
+import mojo from '../lib/mojo.js';
+import jsonConfigPlugin from '../lib/plugins/json-config.js';
+import t from 'tap';
+
+t.test('JSONConfig app', async t => {
+  const client = await app.newTestClient({tap: t});
+
+  await t.test('Named config file', async t => {
+    t.same(app.config, {name: 'Bond. James Bond', drink: 'Martini', car: 'Aston Martin'});
+    (await client.getOk('/')).statusIs(200).bodyIs('My name is Bond. James Bond.');
+  });
+  await client.stop();
+
+  t.test('Default JSONConfig App', async t => {
+    const app = mojo();
+    app.config = {name: 'overridden', extra: 'option'};
+    app.plugin(jsonConfigPlugin, {file: File.currentFile().dirname().child('support', 'named.json').toString()});
+    t.same(app.config, {name: 'namedConfig', extra: 'option', deep: {option: ['deep']}});
+    t.end();
```

Comment:
I think you're missing a test case here where you change the mode of the app and do not find a mode specific config and a value is not redefined from the main config.